src/brushes/base_brush.class.ts

```diff
@@ -75,6 +78,14 @@ export abstract class BaseBrush {
     this.canvas = canvas;
   }
 
+  abstract _render(): void;
+  abstract onMouseDown(pointer: Point, ev: TBrushEventData): void;
+  abstract onMouseMove(pointer: Point, ev: TBrushEventData): void;
+  /**
+   * @returns true if brush should continue blocking interaction
+   */
+  abstract onMouseUp(ev: TBrushEventData): boolean | void;
```

Comment:
I want to make all signature the same
```suggestion
  abstract onMouseUp(pointer: Point, ev: TBrushEventData): boolean | void;
```

Reply:
Let's plan the new api directly.